src/tests/ftest/telemetry/engine_events.py

```diff
@@ -0,0 +1,270 @@
+        events_dead_ranks = [None for _ in range(rank_count)]
+        hosts = list(self.hostlist_servers)
+        for host in hosts:
+            engine_events_dead_ranks = metric_to_data["engine_events_dead_ranks"]
+            metrics = engine_events_dead_ranks[host]["engine_events_dead_ranks"]["metrics"]
+            for metric in metrics:
+                rank = int(metric["labels"]["rank"])
+                events_dead_ranks[rank] = metric["value"]
+
+        events_last_event_ts = [None for _ in range(rank_count)]
+        for host in hosts:
+            engine_events_last_event_ts = metric_to_data["engine_events_last_event_ts"]
+            metrics = engine_events_last_event_ts[host]["engine_events_last_event_ts"]["metrics"]
+            for metric in metrics:
+                rank = int(metric["labels"]["rank"])
+                events_last_event_ts[rank] = metric["value"]
+
+        servicing_at = [None for _ in range(rank_count)]
+        for host in hosts:
+            metrics = metric_to_data["engine_servicing_at"][host]["engine_servicing_at"]["metrics"]
+            for metric in metrics:
+                rank = int(metric["labels"]["rank"])
+                servicing_at[rank] = metric["value"]
+
+        started_at = [None for _ in range(rank_count)]
+        for host in hosts:
+            metrics = metric_to_data["engine_started_at"][host]["engine_started_at"]["metrics"]
+            for metric in metrics:
+                rank = int(metric["labels"]["rank"])
+                started_at[rank] = metric["value"]
+
+        self.log.info("events_dead_ranks = %s", events_dead_ranks)
+        self.log.info("events_last_event_ts =%s", events_last_event_ts)
+        self.log.info("servicing_at = %s", servicing_at)
+        self.log.info("started_at = %s", started_at)
+
+        return (events_dead_ranks, events_last_event_ts, servicing_at, started_at)
+
+    def check_rank_stopped(self, restart_rank, dmg_command):
+        """Check the restart_rank is stopped.
+
+        If it hasn't been stopped, fail the test. This method is created to avoid the pylint error.
+
+        Args:
+            restart_rank (int): Restarted rank.
+            dmg_command (DmgCommand): DmgCommand object.
+        """
+        rank_stopped = False
+        for count in range(3):
+            time.sleep(5)
+            query_out = dmg_command.system_query()
+            for member in query_out["response"]["members"]:
+                if member["rank"] == restart_rank:
+                    if member["state"] == "stopped" or member["state"] == "excluded":
+                        rank_stopped = True
+                        break
+            if rank_stopped:
+                self.log.info("Rank %d is stopped. count = %d", restart_rank, count)
+                break
+            self.log.info("Rank %d is not stopped. Check again. count = %d", restart_rank, count)
+        if not rank_stopped:
+            self.fail(f"Rank {restart_rank} didn't stop!")
```

Comment:
If I'm reading this correctly, I think it could be simplified to this. Since if the rank is stopped, we break out of both loops
```suggestion
        for count in range(3):
            time.sleep(5)
            query_out = dmg_command.system_query()
            for member in query_out["response"]["members"]:
                if member["rank"] == restart_rank:
                    if member["state"] in ("stopped", "excluded"):
                        self.log.info("Rank %d is stopped. count = %d", restart_rank, count)
                        return
            self.log.info("Rank %d is not stopped. Check again. count = %d", restart_rank, count)
        self.fail(f"Rank {restart_rank} didn't stop!")
```

Reply:
Right, this is simpler. Thanks.